### PR TITLE
LVRUBYM-353: Add upload action to teacher

### PR DESCRIPTION
**Pull request by skrypnyk88**
## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/LVRUBYM-353)

## Code reviewers

- [x] @vikabelous
- [x] @yaremakleputs 
- [x] @Mykhailo123

### Second Level Review

- [ ] @omaik

## Summary of issue

Need to add upload action to teacher

## Summary of change

- added upload action to teacher

## Testing approach

We do not need any tests here

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/controllers/v1/teachers_controller.rb [hunk 1]
```diff
@@ -21,6 +21,16 @@ def teacher_params
                                       :locale)
     end
 
+    def upload
```
Comment (vikabelous) [inline app/controllers/v1/teachers_controller.rb]: I have the same method in Students controller. Maybe it's better to move this method to attachable module

Comment (skrypnyk88) [inline app/controllers/v1/teachers_controller.rb]: :+1:

File: .rubocop.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ Metrics/BlockLength:
 Metrics/MethodLength:
   Exclude:
     - 'db/migrate/*'
+    - 'spec/controllers/v1/teachers_controller_spec.rb'
```
Comment (omaik) [inline .rubocop.yml]: I think you can squeeze method in this file

File: app/controllers/concerns/v1/attachable.rb [hunk 3]
```diff
@@ -9,5 +9,15 @@ def attachment_uploader
     def attachment_params
       params.require(:file).permit(:filetype, :filename, :base64)
     end
+
+    def upload_attachment(owner)
```
Comment (omaik) [inline app/controllers/concerns/v1/attachable.rb]: The thing to render can be different in some cases, I think you should pass it as parameter

File: .rubocop.yml (hunk 2 above)
Comment (skrypnyk88) [inline .rubocop.yml]: Yes, but then string will be too long. (86 chars)

File: app/controllers/concerns/v1/attachable.rb (hunk 3 above)
Comment (skrypnyk88) [inline app/controllers/concerns/v1/attachable.rb]: Yes. And I pass owner as parameter which will be render.

File: app/controllers/concerns/v1/attachable.rb [hunk 4]
```diff
@@ -9,5 +9,15 @@ def attachment_uploader
     def attachment_params
       params.require(:file).permit(:filetype, :filename, :base64)
     end
+
+    def upload_attachment(owner, rend_obj)
```
Comment (omaik) [inline app/controllers/concerns/v1/attachable.rb]: shortening of the words only complicates reading, do not do this.

File: app/controllers/v1/students_controller.rb [hunk 5]
```diff
@@ -38,13 +38,7 @@ def destroy
...
 
     def upload
-      attachment = attachment_uploader.call(attachment_owner: @student)
-      if attachment.valid?
-        render @student, status: :created
-      else
-        render json: { errors: attachment.errors.full_messages },
-               status: :bad_request
-      end
+      upload_attachment(@student, @student)
```
Comment (omaik) [inline app/controllers/v1/students_controller.rb]: this looks pretty bad.

File: spec/controllers/v1/teachers_controller_spec.rb [hunk 6]
```diff
@@ -4,32 +4,36 @@
...
 
   def teacher_params(teacher)
-    teacher.attributes
-           .with_indifferent_access
-           .extract!(:id, :first_name, :last_name, :email, :phone, :locale)
+    params =
+      teacher.attributes.with_indifferent_access
+             .extract!(:id, :first_name, :last_name, :email, :phone, :locale)
+
+    if teacher.attachment
```
Comment (omaik) [inline spec/controllers/v1/teachers_controller_spec.rb]: you can still squeeze this operator

File: app/controllers/v1/students_controller.rb (hunk 5 above)
Comment (skrypnyk88) [inline app/controllers/v1/students_controller.rb]: Can I leave only one?

File: app/controllers/concerns/v1/attachable.rb (hunk 4 above)
Comment (skrypnyk88) [inline app/controllers/concerns/v1/attachable.rb]: Ok

File: app/controllers/v1/students_controller.rb (hunk 5 above)
Comment (skrypnyk88) [inline app/controllers/v1/students_controller.rb]: Yes. You are right)))

**Pull request merged**